### PR TITLE
Add new client_rekey resource

### PR DESCRIPTION
**Pull request by lamont-granquist**
Based on https://github.com/chef-boneyard/client-rekey only with none of the legacy, written as a custom resource.

### CODE REVIEW

File: lib/chef/resource/rekey.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class Chef
+  class Resource
+    class Rekey < Chef::Resource
+      provides :rekey
+      resource_name :rekey
```
Comment (tas50) [inline lib/chef/resource/rekey.rb]: seems like this would make more sense if we named it client_rekey

Comment (lamont-granquist) [reply lib/chef/resource/rekey.rb]: yeah that makes sense, will do.

**Comment by tas50** [comment]
@lamont-granquist what's left here before we can ship this?

**Comment by tas50** [comment]
@lamont-granquist anything else that needed to be done to this one?

**Comment by lamont-granquist** [comment]
yeah still needed decent manual testing i think which i ran into snags trying to do

**Comment by lamont-granquist** [comment]
(or decent non-manual testing but that would require a chef-zero that could do auth or spinning up a chef-server for testing)

File: lib/chef/resource/client_rekey.rb [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      description "The client_rekey resource regenerates the key used by #{Chef::Dist::PRODUCT} to communicate with the #{Chef::Dist::SERVER_PRODUCT}."
+      introduced "15.6"
+
+      examples <<~DOC
+        Rekey the node
+        ```ruby
+        client_rekey
+        ```
+
+        Rekey the node with a custom key path
```
Comment (NoamDolovichWS) [inline lib/chef/resource/client_rekey.rb]: noam

**Comment by tas50** [comment]
Closing this one out for now since we don't have a way to test this and we shouldn't pull it in without that.

**Pull request closed without merging**